addons/partner_autocomplete/models/res_company.py

```diff
@@ -1,14 +1,128 @@
 
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def iap_enrich_auto(self):
+        """ Enrich company. This method should be called by automatic processes
+        and a protection is added to avoid doing enrich in a loop. """
+        if self.env.user.has_group('base.group_system'):
+            for company in self.filtered(lambda company: not company.iap_enrich_auto_done):
+                company._enrich()
+            self.iap_enrich_auto_done = True
+        return True
+
+    def _enrich(self):
+        """
+        This method calls the partner autocomplete service from IAP to enrich the partner related fields
+        of the company.
+        """
+        self.ensure_one()
+        _logger.info("Starting enrich of company %s (%s)", self.name, self.id)
+
+        company_domain = self._get_company_domain()
+        if not company_domain:
+            return {'error': _("Could not extract company domain. Impossible to enrich the company.")}
+
+        company_data = self.env['res.partner'].enrich_company(company_domain, False, self.vat, timeout=COMPANY_AC_TIMEOUT)
+        if company_data.get('error'):
+            return {'error': company_data.get('error_message', _('Unkown reason'))}
+
+        additional_data = company_data.pop('additional_info', False)
+
+        if company_data.get('logo'):
+            # Erase image_1920 even if something is in it. Indeed as partner_autocomplete is probably installed as a
+            # core app (mail -> iap -> partner_autocomplete auto install chain) it is unlikely that people already
+            # updated their company logo.
+            try:
+                company_data['image_1920'] = base64.b64encode(
+                    requests.get(company_data['logo'], timeout=COMPANY_AC_TIMEOUT).content
+                )
+            except Exception as e:
+                _logger.warning('Download of image for new company %r failed, error %r' % (self.name, e))
+
+        # Keep only truthy values that are not already set on the target partner
+        company_data = {field: value for field, value in company_data.items()
+                        if field in self.partner_id._fields and field != 'image_1920' and value and not self.partner_id[field]}
+
+        company_data = self._enrich_extract_ids(company_data, ['state_id', 'country_id'])
+        if company_data.get('child_ids'):
+            child_ids = []
+            for child_data in company_data['child_ids']:
+                child_ids.append(self._enrich_extract_ids(child_data, ['state_id', 'country_id']))
+            company_data['child_ids'] = child_ids
+
+        # handle o2m values. Returned in the form: E.g. {'child_ids': [{'name': 'Value 1'}, {'name': 'Value 2'}]}
+        for o2m_field, create_values in company_data.items():
+            if isinstance(create_values, list):
+                commands = [(
+                    0, 0, create_value
```

Comment:
@Julien00859 has worked very hard on [odoo.fields.Command](https://github.com/odoo/odoo/blob/master/odoo/fields.py#L2835). Not using `Command` makes juc very sad. Please don't make juc sad.

Reply:
Replacing 0, 0 that I use since 9 years by "yet another keyword I will forget everytime I use it"  ?

Reply:
Yes.

Reply:
I will wait 9 more years to do it then.

Reply:
![beware the pylint extension](https://www.commitstrip.com/wp-content/uploads/2013/02/Strips-Indentation-600-finalenglish.jpg)

Reply:
`(0, 0, {...})` becomes `Command.create({...})`

Reply:
0, 0 is shorter.